[CI] RandomizedTimeSeriesIT testRateGroupBySubset failing

**Build Scans:**
- [elasticsearch-intake #37581 / part3](https://gradle-enterprise.elastic.co/s/j3pwnz72qexz4)
- [elasticsearch-pull-request #136058 / part-3](https://gradle-enterprise.elastic.co/s/vgqhikhlqxkf2)
- [elasticsearch-pull-request #136058 / part-3](https://gradle-enterprise.elastic.co/s/6pdsxffsfecrq)
- [elasticsearch-pull-request #136058 / part-3](https://gradle-enterprise.elastic.co/s/jl4g35f5am5ai)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT.testRateGroupBySubset" -Dtests.seed=86AA6FB3CDC0EFA7 -Dtests.locale=jgo-Latn-CM -Dtests.timezone=America/Cancun -Druntime.java=26
```

**Applicable branches:**
9.3

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.RandomizedTimeSeriesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRateGroupBySubset'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failed. Agg: INCREASE | Failed windows: 1 windows(0.060024009603841535%):
ROWS: 1666|Failed for row:
[1, 0.0, 0.0, 0.0, 0.0, 0.0, 2025-07-31T09:44:00.000Z, 7a5c:7926::f853:c98e:ffff:ffff, GxkOcDMVnPRq, 939, LYFrbp]
Wanted: RateStats[count=1, max=RateRange[lower=210.6, upper=556.4202334630351], avg=RateRange[lower=210.6, upper=556.4202334630351], min=RateRange[lower=210.6, upper=556.4202334630351], sum=RateRange[lower=210.6, upper=556.4202334630351]]
Exception: 
Expected: (a value less than or equal to <556.4202334630351> and not a value less than <210.6>)
     but: not a value less than <210.6> was <0.0>
Row times and values:
	TS:2025-07-31T09:44:00Z=234.0
```

**Issue Reasons:**
- [9.3] 1 consecutive failures in test testRateGroupBySubset
- [9.3] 1 consecutive failures in step part3
- [9.3] 1 consecutive failures in pipeline elasticsearch-intake
- [9.3] 4 failures in test testRateGroupBySubset (1.8% fail rate in 221 executions)
- [9.3] 3 failures in step part-3 (10.3% fail rate in 29 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/RandomizedTimeSeriesIT.java
@@ -331,6 +331,11 @@ public class RandomizedTimeSeriesIT extends AbstractEsqlIntegTestCase {
                     && timeseries.getFirst().v2().v1().toEpochMilli() % (secondsInWindow * 1000L) == 0
                     && offset > 0) {
                     // Value at lower boundary is present, check if there's one in the previous window to use.
+                    // For INCREASE, return 0 because the increase was already accounted for in the previous bucket.
+                    // For RATE, we still need to calculate the rate using interpolation from the previous bucket.
+                    if (deltaAgg.equals(DeltaAgg.INCREASE)) {
+                        return new RateRange(0.0, 0.0);
+                    }
                     addLastTupleFromLowerWindow(timeseries, allTimeseries.get(offset - 1), secondsInWindow);
                 }
                 if (timeseries.size() < 2) {

